arm-software/linux/docs/OpenMPSupport.md

```diff
@@ -0,0 +1,74 @@
+- **N** : No. When the implementation is absent
+
+Note: No distinction is made between the support in Parser/Semantics, MLIR, Lowering or the OpenMPIRBuilder.
+
+## OpenMP 4.0
+
+| Feature                                                    | Status | Comments                                                |
+|------------------------------------------------------------|--------|---------------------------------------------------------|
+| proc_bind clause                                           | Y      | |
+| simd construct                                             | P      | Some clauses are not supported |
+| declare simd construct                                     | N      | |
+| do simd construct                                          | Y      | |
+| target data construct                                      | N      | |
+| target construct                                           | N      | |
+| target update construct                                    | N      | |
+| declare target directive                                   | N      | |
+| teams construct                                            | N      | |
+| distribute construct                                       | N      | |
+| distribute simd construct                                  | N      | |
+| distribute parallel loop construct                         | N      | |
+| distribute parallel loop simd construct                    | N      | |
+| depend clause                                              | P      | Depend clause with array sections are not supported |
+| declare reduction construct                                | N      | |
+| atomic construct extensions                                | Y      | |
+| cancel construct                                           | N      | |
+| cancellation point construct                               | N      | |
+| parallel do simd construct                                 | Y      | |
+| target teams construct                                     | N      | |
+| teams distribute construct                                 | N      | |
+| teams distribute simd construct                            | N      | |
+| target teams distribute construct                          | N      | |
+| teams distribute parallel loop construct                   | N      | |
+| target teams distribute parallel loop construct            | N      | |
+| teams distribute parallel loop simd construct              | N      | |
+| target teams distribute parallel loop simd construct       | N      | |
+
+## OpenMP 3.1, OpenMP 3.0
+
+| Feature                                                    | Status | Comments                                                |
+|------------------------------------------------------------|--------|---------------------------------------------------------|
+| intent(in) in firstprivate                                 | Y      | |
+| const-qualified types in firstprivate                      | N      | |
```

Comment:
Where does this come from? acfl, gfortran, and llvm-flang main also don't allow this. Testing with
```
integer, parameter :: i = 1

!$omp parallel firstprivate(i)
  print *,i
!$omp end paralllel
end
```

Reply:
I think what this reffers to is on page 32 of OpenMP 3.0
> A variable that appears in a private clause must not have a const-qualified type
unless it is of class type with a mutable member.

Is changed to the following on page 98 of OpenMP 3.1
> A variable that appears in a private clause must not have a const-qualified type
unless it is of class type with a mutable member. This restriction does not apply to
the firstprivate clause

Both statements do not apply to Fortran.

Reply:
OK. I will remove.

In OpenMP 5.1 and later the following appears.
```
Named constants may be listed in a firstprivate clause.
```